Frameworks/CoreText/DWriteWrapper_CoreText.mm

```diff
@@ -802,4 +803,140 @@ CFStringRef _DWriteFontCopyName(const ComPtr<IDWriteFontFace>& fontFace, CFStrin
+                       L"IDWriteGeometrySink::BeginFigure called while a figure was currently in progress. Placing object in error state - "
+                       L"future function calls will fail.");
+            _invalidState = true;
+        }
+
+        if (!_invalidState) {
+            // figureBegin is ignored for CGPath purposes -
+            // filled vs hollow maps to CGPathDrawingMode, and is specified to CGContextDrawPath directly
+            _figureInProgress = true;
+            CGPathMoveToPoint(_cgPath.get(), _transform, startPoint.x, -startPoint.y);
+        }
+    }
+
+    HRESULT STDMETHODCALLTYPE Close() {
+        if (_figureInProgress) {
+            TraceError(TAG,
+                       L"IDWriteGeometrySink::Close called while a figure was currently in progress. Placing object in error state - "
+                       L"future function calls will fail.");
+            _invalidState = true;
+        }
+
+        return _invalidState ? E_INVALIDARG : S_OK;
+    }
+
+    void STDMETHODCALLTYPE EndFigure(D2D1_FIGURE_END figureEnd) {
+        if (!_figureInProgress) {
+            TraceError(TAG,
+                       L"IDWriteGeometrySink::EndFigure called while no figure was currently in progress. Placing object in error state - "
+                       L"future function calls will fail.");
+            _invalidState = true;
+        }
+
+        if (!_invalidState) {
+            // figureBegin is ignored for CGPath purposes -
+            // filled vs hollow maps to CGPathDrawingMode, and is specified to CGContextDrawPath directly
+            _figureInProgress = false;
+
+            if (figureEnd) {
+                // D2D1_FIGURE_END_CLOSED  = 1, close the subpath
+                CGPathCloseSubpath(_cgPath.get());
+            }
+            // D2D1_FIGURE_END_OPEN = 0, subpath is left open (no-op in terms of CGPath)
+        }
+    }
+
+    void STDMETHODCALLTYPE SetFillMode(D2D1_FILL_MODE fillMode) {
+        // No-op for CGPath purposes - CGPathDrawingMode is specified to CGContextDrawPath directly
+    }
+
+    void STDMETHODCALLTYPE SetSegmentFlags(D2D1_PATH_SEGMENT vertexFlags) {
+        // No-op for CGPath purposes - CGLineJoin is specified to CGContextStrokePath directly
+    }
+
+    // Releases the class's ownership of its path when all operations are finished, returning a +1 reference (from the path's Create)
+    CGPathRef ReleasePath() {
+        return _cgPath.release();
+    }
+
+private:
+    woc::unique_cf<CGMutablePathRef> _cgPath;
+    const CGAffineTransform* _transform;
+    bool _figureInProgress = false; // Keeps track of whether this class is currently between a BeginFigure and EndFigure call
+    bool _invalidState = false; // If BeginFigure and EndFigure calls are imbalanced, invalidate all future operations
+};
+
+CGPathGeometrySink::CGPathGeometrySink(const CGAffineTransform* transform) {
+    _cgPath.reset(CGPathCreateMutable());
+    _transform = transform;
```

Comment:
Transform is stored by pointer, thereby requiring the caller manages its lifetime.  could you make a copy here instead, so this class can be free-standing?